api/v1alpha1/types.proto

```diff
@@ -82,6 +82,42 @@ enum HealthState {
 message Health {
   HealthState status = 1;
   repeated string issues = 2;
+  repeated ArgoCDAppState argocd_apps = 3;
+}
+
+message ArgoCDAppState {
+  string namespace = 1;
+  string name = 2;
+  ArgoCDAppHealthStatus health_status = 3;
+  ArgoCDAppSyncStatus sync_status = 4;
+}
+
+message ArgoCDAppHealthStatus {
+  ArgoCDAppHealthState status = 1;
+  string message = 2;
+}
+
+enum ArgoCDAppHealthState {
+  // buf:lint:ignore ENUM_ZERO_VALUE_SUFFIX
+  ARGO_CD_APP_HEALTH_STATE_UNKNOWN = 0;
+  ARGO_CD_APP_HEALTH_STATE_PROGRESSING = 1;
+  ARGO_CD_APP_HEALTH_STATE_HEALTHY = 2;
+  ARGO_CD_APP_HEALTH_STATE_SUSPENDED = 3;
+  ARGO_CD_APP_HEALTH_STATE_DEGRADED = 4;
+  ARGO_CD_APP_HEALTH_STATE_MISSING = 5;
+}
```

Comment:
Do we need to re-evaluate this approach in light of https://github.com/akuity/kargo/pull/714?

Reply:
Yes... fixed it during rebase.